### PR TITLE
feat(pubsub): add attributes sample and cleanup region tags

### LINKED ISSUE #1734: Pub/Sub: samples audit and cleanup
Please rename:
- `pubsub_publish_with_error_handling_that_scales` to `pubsub_publish_with_error_handler`

Please remove:
- the region tag `pubsub_publish` but keep the region tag `pubsub_quickstart_publisher` and the sample.

Please add: 
- `pubsub_subscriber_async_pull_custom_attributes` ([docs](https://cloud.google.com/pubsub/docs/pull#pull-messages-with-custom-attributes))
- `pubsub_publisher_retry_settings`

### PR DESCRIPTION
**Pull request by hongalex**
Clean up samples tags and added a pull messages with custom attributes sample.

Fixes #1734

### CODE REVIEW

**Review by anguillanneuf — APPROVED** [review]

File: pubsub/subscriptions/subscription_test.go [hunk 1]
```diff
@@ -308,6 +308,47 @@ func TestPullMsgsConcurrencyControl(t *testing.T) {
...
+	})
+	if _, err := res.Get(ctx); err != nil {
+		t.Fatalf("Get publish result: %v", err)
+	}
+
+	buf := new(bytes.Buffer)
+	if err := pullMsgsCustomAttributes(buf, tc.ProjectID, subIDAttributes); err != nil {
+		t.Fatalf("failed to pull messages: %v", err)
+	}
+	// Check for number of newlines, which should correspond with number of messages.
```
Comment (tbpg) [inline pubsub/subscriptions/subscription_test.go:346]: Doesn't look like this is counting newlines?

File: pubsub/subscriptions/subscription_test.go [hunk 2]
```diff
@@ -308,6 +308,47 @@ func TestPullMsgsConcurrencyControl(t *testing.T) {
...
+	if !strings.Contains(buf.String(), "foo = bar") {
+		t.Fatalf("pullMsgsCustomAttributes, expected 'foo = bar'\ngot: %s", buf.String())
```
Comment (tbpg) [inline pubsub/subscriptions/subscription_test.go:348]:
```suggestion
	if got, want := buf.String(), "foo = bar"; !strings.Contains(got, want) {
		t.Fatalf("pullMsgsCustomAttribute\ngot: %s\nwant: %q", got, want)
```

https://github.com/golang/go/wiki/TestComments#got-before-want

File: pubsub/topics/publish.go [hunk 3]
```diff
@@ -14,7 +14,6 @@
 
 package topics
 
-// [START pubsub_publish]
```
Comment (tbpg) [inline pubsub/topics/publish.go:17]: Confirming docs are updated to not depend on deleted region tags.

**Review by tbpg — CHANGES_REQUESTED** [review]

File: pubsub/topics/publish.go (hunk 3 above)
Comment (hongalex) [reply pubsub/topics/publish.go:17]: Yeah, just confirmed in devrel/samples we only depend on the quickstart ones.

**Review by tbpg — APPROVED** [review]

**Pull request merged**